map_model/src/traversable.rs

```diff
@@ -245,17 +246,118 @@ impl Traversable {
     }
 }
 
-// 10mph
+// 10 mph
 pub const MAX_BIKE_SPEED: Speed = Speed::const_meters_per_second(4.4704);
-// 3mph
-pub const MAX_WALKING_SPEED: Speed = Speed::const_meters_per_second(1.34);
+// 3 mph
+pub const MAX_WALKING_SPEED: Speed = Speed::const_meters_per_second(1.34112);
+
+fn bike_speed_on_incline(max_speed: Speed, percent_incline: f64) -> Speed {
+    // There doesn't seem to be a straightforward way of calculating how an "average" cyclist's
+    // speed is affected by hills. http://www.kreuzotter.de/english/espeed.htm has lots of detail,
+    // but we'd need to guess values like body size, type of bike, etc.
+    // https://github.com/ibi-group/OpenTripPlanner/blob/65dcf0a4142e31028cf9d1b2c15ad32dd1084252/src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java#L934-L1082
+    // is built from this, but seems to be more appropriate for motorized micromobility devices
+    // like e-scooters.
+
+    // So, we'll adapt the table from Valhalla --
+    // https://valhalla.readthedocs.io/en/latest/sif/elevation_costing/ describes how this works.
+    // Their "weighted grade" should be roughly equivalent to how the elevation_lookups package we
+    // use calculates things.  This table comes from
+    // https://github.com/valhalla/valhalla/blob/f899a940ccbd0bc986769197dec5bb9383014afb/src/sif/bicyclecost.cc#L139.
+    // Valhalla is MIT licensed: https://github.com/valhalla/valhalla/blob/master/COPYING.
+
+    // TODO Could binary search or do something a bit faster here, but doesn't matter much
+    let pct = percent_incline * 100.0;
+    for (grade, factor) in vec![
+        (-10.0, 2.2),
+        (-8.0, 2.0),
+        (-6.5, 1.9),
+        (-5.0, 1.7),
+        (-3.0, 1.4),
+        (-1.5, 1.2),
+        (0.0, 1.0),
+        (1.5, 0.95),
+        (3.0, 0.85),
+        (5.0, 0.75),
+        (6.5, 0.65),
+        (8.0, 0.55),
+        (10.0, 0.5),
```

Comment:
Not for this PR, but this might be a good opportunity to surface some agent decoration - some reason to explain why they're bike sooooo slow, especially since our rendering is all d2.

Reply:
Definitely! A thought bubble with some kind of sweaty struggle and a triangle of different slopes would look great.